crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -2400,6 +2400,71 @@ impl Backend {
             .lock()
             .retain(|tx| tx.unbounded_send(notification.clone()).is_ok());
     }
+
+    /// Reorg the chain to a common height and execute blocks to build new chain
+    pub async fn reorg(
+        &self,
+        depth: u64,
+        tx_pairs: HashMap<u64, Vec<Arc<PoolTransaction>>>,
```

Comment:
do we need the hashmap here or can this be a Vec of Vec?

Reply:
Opted for `&U64::from(i)` here.

Reply:
We will need a hashmap here incase there is a gap between block tx pairs, e.g. where `depth=5`, tx's could be mined into blocks 1 and 3.